### ISSUE TITLE
Roundtripping CMODs fails on old versions of fmt (<6.1.0)

### ISSUE THREAD

**Issue by ajtribick** [CelestiaProject/Celestia#1179]
Investigating the build failure on Ubuntu 18.04 for the CMOD roundtrip test in #1178, I decided to check the output behaviour of various fmt versions. Test code:
```C++
#include <cmath>
#include <cstdint>
#include <cstring>
#include <fmt/format.h>
int main()
{
    float f;
    std::uint32_t i = 0;
    do
    {
        std::memcpy(&f, &i, sizeof(float));
        if (!std::isfinite(f)) { continue; }
        auto s = fmt::format("{}", f); // line A
        float g = std::strtof(s.data(), nullptr);
        if (f != g)
        {
            fmt::print("Error detected! i={}\n", i);
            break;
        }
    } while (++i != 0);
}
```
This fails with fmt 4.0.0, the version supplied by Ubuntu 18.04. Consistent with the release notes of fmt, the earliest version that passes is 6.1.0, which defaults to the shortest representation capable of representing the value (which in my opinion is the reasonable default behaviour).

One option would be to explicitly use the format parameter `{:.9g}` with `float` (the 9 is the same value as returned from `std::numeric_limits<float>::max_digits10`), this produces roundtrippable float representations on 4.0.0 at the expense of producing some overlong representations on fmt >6.1.0, e.g. on version 6.1.0, a value of `0.3f` with the default format produces "0.3", while using `{:.9g}` gives "0.300000012".

The other option would be to explicitly include a version of fmt ≥6.1.0 and fall back to that if the system provides a version of fmt that is too old. Advantage is, we get what is probably the most reasonable output, and we don't have to hunt down all the various places in the code where we're outputting floats via fmt.

Either way, we should probably specify a minimum version of the fmt library we're prepared to use.

**Comment by 375gnu** [comment]
Let's bump minimal supported fmt to 6.1.

**Comment by ajtribick** [comment]
Closed via #1178
